src/App.jsx

```diff
@@ -2,12 +2,24 @@ import "./App.css";
 import TodoTemplate from "./components/TodoTemplate";
 import TodoInsert from "./components/TodoInsert";
 import TodoList from "./components/TodoList";
-import { useState } from "react";
+import { useCallback, useState } from "react";
+
+function createBulkTodos() {
+  const array = [];
+  for (let i = 1; i < 7500; i++) {
+    array.push({
+      id: i,
+      text: `할 일${i}`,
+      checked: false,
+    });
+  }
+  return array;
+}
 
 function App() {
-  const [todoList, setTodoList] = useState([]);
+  const [todoList, setTodoList] = useState(() => createBulkTodos());
 
-  const handleAddTodoItem = (item) => {
+  const handleAddTodoItem = useCallback((item) => {
```

Comment:
```javascript
 const handleAddTodoItem = useCallback((item) => {
    setTodoList((prev) => {
      const nextId = prev.length > 0 ? prev[prev.length - 1].id + 1 : 1;
      return [
        ...prev,
        {
          id: nextId,
          text: item,
          checked: false,
        },
      ];
    });
  };
  }, []);
```
를 보면 `prev`를 기반으로 id를 계산하고 있죠! 순수함수 형태를 만들어 불변성 유지를 위해 배열을 새로 복사하고 있습니다! 근데 생각해보시면 이 형태는 실행될 때마다 성능면에서 O(n)을 가지게 되어요. 길이가 길면 비용이 커질 수 있습니다! 또, 정렬에도 민감하죠!
만약 저라면 `useRef`와 `concat`을 사용할 것 같아요.
```suggestion
 const nextId = useRef(1);

const onInsert = useCallback((text) => {
  const todo = { id: nextId.current, text, checked: false };
  setTodos(todos => todos.concat(todo));
  nextId.current += 1;
}, []);
```
이러면 setTodos에서 prev => 사용해서 기존 상태에 안전하게 접근도 가능하고 concat이 성능면에서 더 우수하기 때문에 이 방법을 쓸 것 같습니다! 다만 ID생성이 컴포넌트의 외부 상태에 의존한다는 단점이 있어요..! 
제 제안이 정답은 아니니 창우님께서 비교해보시고 어느점이 더 낫다고 생각하는지 공유해주시면 더 좋을 것 같네요! 👍🏻

Reply:
아.. 정렬되는 경우를 생각 못했네요. 다시 보니 정렬등에 의해서 배열이 바뀔 경우 마지막 id값이 제일 큰 값이 아니게 될 수도 있군요.
그리고 `concat`을 사용하지 않았어서 몰랐는데, 배열이 아닌 값을 추가하거나 할 때에도 이점이 있네요.
적어도 제가 한 방법보다 확실히 나은 방법이라고 생각합니다 적용해볼게요!